generated/nidaqmx/base_interpreter.py

```diff
@@ -0,0 +1,1630 @@
+            pause_when=Level.HIGH,
+            ready_event_active_level=Polarity.ACTIVE_HIGH):
+        pass
+
+    def cfg_burst_handshaking_timing_import_clock(
+            self, sample_clk_rate, sample_clk_src,
+            sample_mode=AcquisitionType.FINITE, samps_per_chan=1000,
+            sample_clk_active_edge=Edge.RISING, pause_when=Level.HIGH,
+            ready_event_active_level=Polarity.ACTIVE_HIGH):
+        pass
+
+    def cfg_change_detection_timing(
+            self, rising_edge_chan="", falling_edge_chan="",
+            sample_mode=AcquisitionType.FINITE, samps_per_chan=1000):
+        pass
+
+    def cfg_dig_edge_ref_trig(
+            self, trigger_source, pretrigger_samples,
+            trigger_edge=Edge.RISING):
+        pass
+
+    def cfg_dig_edge_start_trig(
+            self, trigger_source, trigger_edge=Edge.RISING):
+        pass
+
+    def cfg_dig_pattern_ref_trig(
+            self, trigger_source, trigger_pattern, pretrigger_samples,
+            trigger_when=DigitalPatternCondition.PATTERN_MATCHES):
+        pass
+
+    def cfg_dig_pattern_start_trig(
+            self, trigger_source, trigger_pattern,
+            trigger_when=DigitalPatternCondition.PATTERN_MATCHES):
+        pass
+
+    def cfg_handshaking_timing(
+            self, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000):
+        pass
+
+    def cfg_implicit_timing(
+            self, sample_mode=AcquisitionType.FINITE, samps_per_chan=1000):
+        pass
+
+    def cfg_input_buffer(self):
+        pass
+
+    def cfg_output_buffer(self):
+        pass
+
+    def cfg_pipelined_samp_clk_timing(
+            self, rate, source="", active_edge=Edge.RISING,
+            sample_mode=AcquisitionType.FINITE, samps_per_chan=1000):
+        pass
+
+    def cfg_samp_clk_timing(
+            self, rate, source="", active_edge=Edge.RISING,
+            sample_mode=AcquisitionType.FINITE, samps_per_chan=1000):
+        pass
+
+    def cfg_time_start_trig(self, when, timescale=Timescale.USE_HOST):
+        pass
+
+    def cfg_watchdog_ao_expir_states(self):
+        pass
+
+    def cfg_watchdog_co_expir_states(self):
+        pass
+
+    def cfg_watchdog_do_expir_states(self):
+        pass
+
+    def clear_task(self):
+        pass
+
+    def clear_teds(self, physical_channel):
+        pass
+
+    def configure_logging(self):
+        pass
+
+    def configure_teds(self, physical_channel="", file_path=""):
+        pass
+
+    def connect_terms(
+            self, source_terminal, destination_terminal,
+            signal_modifiers=SignalModifiers.DO_NOT_INVERT_POLARITY):
+        pass
+
+    def control_watchdog_task(self):
+        pass
+
+    def delete_network_device(self):
+        pass
+
+    def delete_saved_global_chan(self):
+        pass
+
+    def delete_saved_scale(self):
+        pass
+
+    def delete_saved_task(self):
+        pass
+
+    def device_supports_cal(self):
+        pass
```

Comment:
Functions that operate on named objects are missing the object name:
- `delete_network_device` and `device_supports_cal` should take a device name.
- `delete_saved_global_chan` should take a global channel name.
- `delete_saved_scale` should take a scale name.
- `delete_saved_task` should take a task name. (Saved tasks are referred to by name, not handle.)

We agreed that the interpreter class will have a field for the task handle (local) or task name (gRPC), so the logic for removing taskHandle parameters is still valid here. However, the logic for removing device names, scale names, etc. does not apply to interpreter methods.

Reply:
Sure. I have added all the parameters from the metadata in the functions signature (Though I have filtered some parameters that doesn't have `type` in the metadata itself).